src/components/ProgramRecord/ProgramRecordActions.jsx

```diff
@@ -0,0 +1,237 @@
+import React, { useState, useEffect } from 'react';
+
+import PropTypes from 'prop-types';
+import {
+  Help, AccountBalance, ContentCopy, Download, SpinnerSimple, Check,
+} from '@edx/paragon/icons';
+import {
+  Button, StatefulButton, Icon, IconButton, OverlayTrigger, Popover, Tooltip, Toast,
+} from '@edx/paragon';
+
+import { FormattedMessage } from '@edx/frontend-platform/i18n';
+import { logError } from '@edx/frontend-platform/logging';
+
+import { getConfig } from '@edx/frontend-platform';
+import { getProgramRecordUrl, getProgramRecordCsv } from './data/service';
+
+const ProgramRecordActions = ({
+  showSendRecordButton, isPublic, toggleSendRecordModal, renderBackButton, username, programId,
+}) => {
+  const [programRecordUrl, setProgramRecordUrl] = useState('');
+  const [showCopyTooltip, setShowCopyTooltip] = useState(false);
+  const [showDownloadToast, setShowDownloadToast] = useState(false);
+  const [downloadRecord, setDownloadRecord] = useState('default');
+
+  const downloadButtonProps = {
+    labels: {
+      default: 'Download program record',
+      pending: 'Downloading program record',
+      complete: 'Download complete',
+    },
+    icons: {
+      default: <Icon src={Download} />,
+      pending: <Icon src={SpinnerSimple} className="icon-spin" />,
+      complete: <Icon src={Check} />,
+    },
+    disabledStates: ['pending', 'complete'],
+    variant: 'primary',
+  };
+
+  useEffect(() => {
+    if (!isPublic) {
+      const handleProgramUrlCopy = async () => {
+        await getProgramRecordUrl(programId, username)
+          .then(({ data }) => {
+            setProgramRecordUrl(data.url);
+          })
+          .catch((error) => {
+            logError(error);
+            throw new Error(error);
+          });
+      };
+      handleProgramUrlCopy();
+    }
+  }, [programId, username, isPublic]);
+
+  useEffect(() => {
+    if (showCopyTooltip) {
+      setTimeout(() => {
+        setShowCopyTooltip(false);
+      }, 3000);
+    }
+  }, [showCopyTooltip]);
+
+  useEffect(() => {
+    if (downloadRecord === 'complete') {
+      setTimeout(() => {
+        setDownloadRecord('default');
+      }, 4000);
+    }
+  }, [downloadRecord]);
+
+  useEffect(() => {
+    // Removes focus from buttons - might not meet accessiblity standards but this matches the legacy behavior
+    document.activeElement.blur();
+  });
+
+  const handleCopyButtonClick = () => {
+    navigator.clipboard.writeText(programRecordUrl);
+    setShowCopyTooltip(true);
+  };
+
+  const handleDownloadRecord = async () => {
+    setDownloadRecord('pending');
+    await getProgramRecordCsv(programId)
```

Comment:
we shouldn't have to await here and create a `Promise ->  Promise`. If all the logic is contained within the promise, the promise itself will resolve and return properly.